include/El/core/DistGraph.hpp

```diff
@@ -11,6 +11,206 @@
 #ifndef EL_CORE_DISTGRAPH_HPP
 #define EL_CORE_DISTGRAPH_HPP
 
-#include <El/core/DistGraph/decl.hpp>
```

Comment:
Can you quantify that extra time to pull in both parts of the file?

Also, I really am referring to splitting headers into the part that only declares the types and the part that contains the API associated with that type. It is common for dependent headers to only need the declaration -- only compilation units that actually manipulate that type need to include the rest of the header. You'll find many such examples in `/usr/include/**/*{fwd,decl,type,types}.h{,pp}`. Boost is insistent about factoring headers in that way, for example.